### PR TITLE
MOB-545 Android: Add preference setting of gas token

### PR DESCRIPTION
**Pull request by ruixhuang**
[untitled.webm](https://github.com/dydxprotocol/v4-native-android/assets/102453770/e1dc5574-9560-4e25-82e1-05e776ae4e47)

### CODE REVIEW

File: v4/feature/shared/src/main/java/exchange/dydx/trading/feature/shared/analytics/RoutingAnalytics.kt [hunk 1]
```diff
@@ -19,7 +19,10 @@ class RoutingAnalytics @Inject constructor(
             pathWithArguments = pathWithArguments.replace("market/{marketId}", "trade/{marketId}")
         }
         arguments?.keySet()?.forEach { key ->
-            pathWithArguments = pathWithArguments.replace("{$key}", arguments.getString(key) ?: "")
+            if (!key.startsWith("android-support-nav:controller:deepLinkIntent")) {
```
Comment (ruixhuang) [inline v4/feature/shared/src/main/java/exchange/dydx/trading/feature/shared/analytics/RoutingAnalytics.kt:22]: Fixing getString() exception

File: v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import exchange.dydx.dydxstatemanager.AbacusStateManagerProtocol
+import exchange.dydx.trading.feature.shared.PreferenceKeys
+import exchange.dydx.utilities.utils.Logging
+import exchange.dydx.utilities.utils.SharedPreferencesStore
+import exchange.dydx.utilities.utils.WorkerProtocol
+import kotlinx.coroutines.CoroutineScope
+
+private const val TAG = "DydxGasTokenWorker"
+
+class DydxGasTokenWorker(
```
Comment (prashanDYDX) [inline v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt:12]: why not inject this?

File: v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    override val scope: CoroutineScope,
```
Comment (prashanDYDX) [inline v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt]: is this used?

File: v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt (hunk 2 above)
Comment (ruixhuang) [reply v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt:12]: We can refactor this later... it involves changes to the parent worker as well.

**Review by prashanDYDX — APPROVED** [review]

**Pull request merged**